src/plugins/s390.c

```diff
@@ -154,13 +154,14 @@ gboolean bd_s390_is_tech_avail (BDS390Tech tech, guint64 mode, GError **error) {
  */
 gboolean bd_s390_dasd_format (const gchar *dasd, const BDExtraArg **extra, GError **error) {
     gboolean rc = FALSE;
-    gchar *dev = NULL;
-    const gchar *argv[8] = {"/sbin/dasdfmt", "-y", "-d", "cdl", "-b", "4096", dev, NULL};
+    const gchar *argv[8] = {"/sbin/dasdfmt", "-y", "-d", "cdl", "-b", "4096", NULL, NULL};
 
     if (!check_deps (&avail_deps, DEPS_DASDFMT_MASK, deps, DEPS_LAST, &deps_check_lock, error))
         return FALSE;
 
-    dev = g_strdup_printf ("/dev/%s", dasd);
+    gchar *dev = g_strdup_printf ("/dev/%s", dasd);
+    argv[6] = dev;
```

Comment:
I don't really see a reason for the ``dev`` variable here. Just use ``argv[6]`` here and the ``g_free()`` call below

Reply:
Fixed.